src/net_components/layers/conv2d.jl

```diff
@@ -97,22 +101,81 @@ function increment!(s::JuMP.AffExpr, input_val::Real, filter_val::JuMP.Variable)
     push!(s, Float64(input_val), filter_val)
 end
 
+function compute_output_parameters(
+    in_height::Int, in_width::Int,
+    filter_height::Int, filter_width::Int, stride::Int,
+    padding::FixedPadding
+)::Tuple{NTuple{2, Int}, NTuple{2, Int}}
+    (top_padding, bottom_padding, left_padding, right_padding) = compute_padding_values(padding)
+    out_height = round(Int, (in_height + top_padding + bottom_padding - filter_height) / stride, RoundDown) + 1
+    out_width = round(Int, (in_width + left_padding + right_padding - filter_width) / stride, RoundDown) + 1
+    
+    output_size = (out_height, out_width)
+    filter_offset = (top_padding, left_padding)
+    return (output_size, filter_offset)
+end
 
+function compute_output_parameters(
+    in_height::Int, in_width::Int,
+    filter_height::Int, filter_width::Int, stride::Int,
+    padding::SamePadding
+)::Tuple{NTuple{2, Int}, NTuple{2, Int}}
+    out_height = round(Int, in_height/stride, RoundUp)
+    out_width = round(Int, in_width/stride, RoundUp)
+    pad_along_height = max((out_height - 1)*stride + filter_height - in_height, 0)
+    pad_along_width = max((out_width - 1)*stride + filter_width - in_width, 0)
+    filter_height_offset = round(Int, pad_along_height/2, RoundDown)
+    filter_width_offset = round(Int, pad_along_width/2, RoundDown)
+    
+    output_size = (out_height, out_width)
+    filter_offset = (filter_height_offset, filter_width_offset)
+    return (output_size, filter_offset)
+end
+
+function compute_output_parameters(
+    in_height::Int, in_width::Int,
+    filter_height::Int, filter_width::Int, stride::Int, 
+    padding::ValidPadding
+)::Tuple{NTuple{2, Int}, NTuple{2, Int}}
+    out_height = round(Int, (in_height + 1 - filter_height) / stride, RoundUp)
+    out_width = round(Int, (in_width + 1 - filter_width) / stride, RoundUp)
+    return((out_height, out_width), (0, 0))
+end
+
+function compute_padding_values(
+    padding::Int
+)::NTuple{4, Int}
+    return (padding, padding, padding, padding)
+end
+
+function compute_padding_values(
+    padding::NTuple{2, Int}
+)::NTuple{4, Int}
+    (y_padding, x_padding) = padding
+    return (y_padding, y_padding, x_padding, x_padding)
+end
+
+function compute_padding_values(
+    padding::NTuple{4, Int}
+)::NTuple{4, Int}
+    return padding
+end
 
 """
 $(SIGNATURES)
 
 Computes the result of convolving `input` with the `filter` and `bias` stored in `params`.
 
-Mirrors `tf.nn.conv2d` from the `tensorflow` package, with `strides = [1, 1, 1, 1].
+Mirrors `tf.nn.conv2d` from the `tensorflow` package, with 
+`strides = [1, params.stride, params.stride, 1]`.
```

Comment:
Incidental change to outdated documentation --- this is unrelated to your PR but I noticed it while I was reviewing my own documentation.